app/assets/src/components/views/PipelineViz/PipelineViz.jsx

```diff
@@ -48,29 +50,98 @@ class PipelineViz extends React.Component {
       sidebarVisible: false,
       sidebarParams: {},
       hovered: false,
+      graphData: props.graphData,
     };
   }
 
   componentDidMount() {
     this.drawGraphs();
     window.addEventListener("resize", this.handleWindowResize);
+    // Only set update loop if the pipeline is still running.
+    if (!this.pipelineIsFinished()) {
+      this.updateLoop = setInterval(() => this.updateGraphs(), 1000);
+    }
   }
 
   componentWillUnmount() {
     window.removeEventListener("resize", this.handleWindowResize);
+    clearInterval(this.updateLoop);
+  }
+
+  async updateGraphs() {
+    const { sample, pipelineRun } = this.props;
+    const { graphData } = this.state;
+
+    const oldGraphData = graphData;
+    const newGraphData = await getGraph(
+      sample.id,
+      pipelineRun.version.pipeline
+    );
+    this.setState({
+      graphData: newGraphData,
+    });
+    const updates = diff(oldGraphData, newGraphData);
+    if (updates.stages) {
+      Object.keys(updates.stages).forEach(stageIndexStr => {
+        const stageIndex = parseInt(stageIndexStr);
+
+        if (stageIndex < this.graphs.length) {
+          // Graph already exists.
+          const graph = this.graphs[stageIndex];
+          const steps = updates.stages[stageIndexStr].steps;
+          Object.keys(steps).forEach(stepIndexStr => {
+            const stepIndex = parseInt(stepIndexStr);
+            const status = steps[stepIndex].status;
+            graph.updateNodes([stepIndex], {
+              group: status,
+              ...this.getNodeStatusOptions(status),
+            });
+          });
+        } else {
+          // A new stage has started and needs to be drawn.
+          if (stageIndex > 0) {
+            // Modify previous stage's graph to create edges to new stage
+            const prevGraph = this.graphs[stageIndex - 1];
+            this.generateEdgeData(stageIndex - 1).forEach(edge => {
+              const { id, ...options } = edge;
+              prevGraph.updateEdges([id], options);
+            });
+          }
+          this.drawStageGraph(stageIndex);
+        }
+      });
+
+      if (this.pipelineIsFinished()) {
```

Comment:
You should add this as callback of `setState`. This would probably be called before the state is updated with the new graph.